simple_repository/model.py

```diff
@@ -138,15 +210,24 @@ def __post_init__(self) -> None:
                 str(safe_version(file.filename, self._normalized_name))
                 for file in self.files
             }
-        else:
-            versions = None
-        object.__setattr__(self, "versions", versions)
+            if self.versions:
+                # If we were given some versions already, confirm that they meet the needs of
+                # the files, per PEP-700.
+                if versions - self.versions:
+                    raise ValueError(
+                        "The versions specified in ProjectDetail does not include all of the "
```

Comment:
Turns out that I get this error when visiting the numpy detail page. Perhaps this should be a warning, even though for some reason it is not following the spec (maybe it is our safe version implementation)